apps/src/weblab/brambleHost.js

```diff
@@ -693,7 +703,7 @@ function load(Bramble) {
     }
 
     bramble.on('inspectorChange', handleInspectorChange);
-    bramble.on('fileChange', handleFileChange);
+    bramble.on('fileChange', validateFileAndHandleChange);
```

Comment:
Will this logic get triggered if a school proxy server is modifying the HTML for the WebLab project as it is being fetched from Code Studio?

Reply:
yes, this event is triggered any time the content of the file changes

Reply:
I know this is already merged, but follow up question ... does this event trigger on initial load of the WebLab project? The scenario I’m thinking of is:
1) student opens existing WebLab project created at home while on school network
2) school network’s HTTP proxy server re-writes the HTML for the project as it’s downloading it to insert script tag
3) student sees (but hasn’t yet tried to edit) their WebLab project with additional script tag in it.

Reply:
it doesn't trigger on initial load, but should trigger in the scenario you're talking about because `fileChange` will be triggered by step 2. i can't mock this scenario perfectly because i don't have the extension that causes this problem, but i've done my best to mock this scenario specifically

Reply:
Do we know, yet, whether it’s a browser extension or a proxy server that’s adding the script tag? With the current solution, if the proxy server is modifying the HTML in-flight as it’s downloading it from our server, then it appears that the student will see the script tag because the file change event has been triggered yet.

Reply:
no, we don't know. we would have to have an affected user confirm whether or not this fix has actually solved the problem as we've never pinpointed the cause or been able to reproduce it

Reply:
[realizing now that you're referring to my "i don't have the extension that causes this" comment, which is inaccurate. i don't have the conditions that cause this, whether its an extension, proxy server, or something else]

Reply:
Totally understand. Either way, this change is going to reduce the impact on students of this content filter 🎉

Reply:
i really hope so! 😅